File: android/src/main/kotlin/com/anisrehman/archarts/ChartViewRegistry.kt
package com.anisrehman.archarts

import java.util.concurrent.ConcurrentHashMap

internal object ChartViewRegistry {
    private val lineCharts = ConcurrentHashMap<Int, LineChartPlatformView>()
    private val barCharts = ConcurrentHashMap<Int, BarChartPlatformView>()

    fun registerLineChart(viewId: Int, view: LineChartPlatformView) {
        lineCharts[viewId] = view
    }

    fun unregisterLineChart(viewId: Int) {
        lineCharts.remove(viewId)
    }

    fun getLineChart(viewId: Int): LineChartPlatformView? = lineCharts[viewId]

    fun registerBarChart(viewId: Int, view: BarChartPlatformView) {
        barCharts[viewId] = view
    }

    fun unregisterBarChart(viewId: Int) {
        barCharts.remove(viewId)
    }

    fun getBarChart(viewId: Int): BarChartPlatformView? = barCharts[viewId]
}

File: ios/ar_charts/Sources/ar_charts/ChartViewRegistry.swift
import Foundation

enum ChartViewRegistry {
    private static let lock = NSLock()
    private static var lineCharts: [Int64: LineChartPlatformView] = [:]
    private static var barCharts: [Int64: BarChartPlatformView] = [:]

    static func registerLineChart(viewId: Int64, view: LineChartPlatformView) {
        lock.lock()
        defer { lock.unlock() }
        lineCharts[viewId] = view
    }

    static func unregisterLineChart(viewId: Int64) {
        lock.lock()
        defer { lock.unlock() }
        lineCharts.removeValue(forKey: viewId)
    }

    static func getLineChart(viewId: Int64) -> LineChartPlatformView? {
        lock.lock()
        defer { lock.unlock() }
        return lineCharts[viewId]
    }

    static func registerBarChart(viewId: Int64, view: BarChartPlatformView) {
        lock.lock()
        defer { lock.unlock() }
        barCharts[viewId] = view
    }

    static func unregisterBarChart(viewId: Int64) {
        lock.lock()
        defer { lock.unlock() }
        barCharts.removeValue(forKey: viewId)
    }

    static func getBarChart(viewId: Int64) -> BarChartPlatformView? {
        lock.lock()
        defer { lock.unlock() }
        return barCharts[viewId]
    }
}
